### DISCUSSION TITLE
[FEATURE REQUEST] file.managed - way it handles http/https source

### DISCUSSION THREAD

**Discussion by lgogolin** [saltstack/salt#67465] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
`file.managed` cache files or download file every time if `keep_source: False` is set.
This is problematic with big files

**Describe the solution you'd like**
To avoid downloading file when `keep_source: False` is set, file.managed should check `Last-Modified` header. 
The same way as `wget -N` is doing it. So at least We would avoid downloading file if it's not needed

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by cmcmarrow** [comment]
Multiple salt user will benefit from having this feature and their network usage should have a noticeable performance boost.

**Comment by cmcmarrow** [comment]
Also welcome to the salt Community @lgogolin!
